base/src/main/java/org/aya/concrete/Expr.java

```diff
@@ -708,15 +708,36 @@ public static Expr.Array newGenerator(
    * <ul>
    *   <li>{@link Let.Bind#bindName} = f</li>
    *   <li>{@link Let.Bind#telescope} = (x : X)</li>
-   *   <li>{@link Let,Bind#result} = G</li>
+   *   <li>{@link Let.Bind#result} = G</li>
    *   <li>{@link Let.Bind#definedAs} = g</li>
    *   <li>{@link Let#body} = expr</li>
    * </ul>
+   *
+   * @param sourcePos to the source pos of body, consider this code
+   *                  <pre>
+   *                  let a := b in<br/>
+   *                  let c := d in<br/>
+   *                  e
+   *                  </pre>
+   *                  The source pos of the
+   *                  {@code let a := b in let c := d in e}
+   *                  and
+   *                  {@code let c := d in e}
+   *                  are both the source pos of e.
+   *                  If you are looking for the source pos of {@code a := b} or {@code c := d},
+   *                  see {@link Let.Bind#sourcePos()}
    */
   record Let(
+    @NotNull SourcePos sourcePos,
     @NotNull Let.Bind bind,
     @NotNull Expr body
   ) implements Expr {
+    public Let {
+      if (body instanceof Let let) {
+        sourcePos = let.sourcePos();
+      }
+    }
```

Comment:
The syntax location should not be the same as the error reporting location. The correct `sourcePos` of a `Let` should always be `sourcePosOf(GenericNode)` in the producer. Or some location-based features may not work freely for `Let`

Reply:
So what do we expect? The whole let-expression or the body?

Reply:
0. Let should have whole SourcePos
1. LetBind can have another SourcePos and we use that for error report

Reply:
So: For all `Let`s in the same group, they share the same source pos -- the whole source pos

Reply:
I think it is good to have a way to obtain the source pos of an expr that determines the type of the whole expr (like the last `DoBind` of a do-notation, the body of `Let`)

Reply:
> So: For all `Let`s in the same group, they share the same source pos -- the whole source pos

No

Reply:
> I think it is good to have a way to obtain the source pos of an expr that determines the type of the whole expr (like the last `DoBind` of a do-notation, the body of `Let`)

Yes